src/irc/BridgedClient.ts

```diff
@@ -442,17 +442,28 @@ export class BridgedClient extends EventEmitter {
         });
     }
 
-    public leaveChannel(channel: string, reason = "User left") {
+    public async leaveChannel(channel: string, reason = "User left"): Promise<void> {
         if (this.state.status !== BridgedClientStatus.CONNECTED) {
-            return Promise.resolve(); // we were never connected to the network.
+            return undefined; // we were never connected to the network.
```

Comment:
Does TypeScript prefer `return undefined` over `return`?

Reply:
Our linter is a pain, if we return a defer.promise then it expects all the others to return something. And yes, eslint makes a distinction between `return undefined` and `return;`;

Reply:
I am wondering whether consistent return is useful given we have types now, so as long as we type the function responses it's probably overkill.